### PR TITLE
Delay fix

### PR DESCRIPTION
**Pull request by tzachi-dar**
This is a very small code refactoring, that is needed in order to make the code more readable.

Should not have any impact on real functionality.

### CODE REVIEW

**Comment by jstevensog** [comment]
OK.  I like some of the changes, particularly getting rid of the redundant channel global variable.  But the wake_before_packet and wake_wixel_before packet are the same thing.  I'd rather keep them as one variable than split into two.  I deliberately didn't use wixel in the name as this code may be ported to another platform with CC251x processor.  Same libraries can be used, just not the wixel specific ones.
I'll keep them as a single variable for now.
Thanks for the suggestions.

**Comment by tzachi-dar** [comment]
I can't seem to see how wake_wixel_before and wake_before_packet are the
same value.

One of them is in milliseconds and the other is in seconds.

More than that, let's look at:
// the current time is greater than the last pkt_time, so we just do a
basic calculation.
// 5 mins in ms + the wake_before_packet time - (current ms - last packet
ms)
delay = (300000 + wake_before_packet) - (getMs() - pkt_time);

If I want wake_before_packet to be bigger, doesn't it mean that I need to
wait for less time?
In other words, I believe that this should be:

delay = (300000 - wake_before_packet) - (getMs() - pkt_time);

Replaced the '+' with a '-'

Thanks
Tzachi
